### PR TITLE
Fix/id map resolution

### LINKED ISSUE #528: Incorrect results when querying schema.org vocabulary
The query:

```clojure
'{:select {?s [:*]}
  :where  [[?s "schema:rangeIncludes" {"@id" "schema:DateTime"}]
           [?s "@id" "schema:inLanguage"]]}
```

executed against a database value containing the schema.org vocabulary should not return results because the `"schema:inLanguage"` subject does not have the `"schema:DateTime"` subject as a value of its `"schema:rangeIncludes"` property. However, this query does return the following result:

```clojure
[{"@id" "schema:inLanguage",
  "rdf:type" ["rdf:Property"],
  "rdfs:comment"
  "The language of the content or performance or used in an action. Please use one of the language codes from the [IETF BCP 47 standard](http://tools.ietf.org/html/bcp47). See also [[availableLanguage]].",
  "rdfs:label" "inLanguage",
  "schema:domainIncludes"
  [{"@id" "schema:CreativeWork"}
   {"@id" "schema:Event"}
   {"@id" "schema:CommunicateAction"}
   {"@id" "schema:WriteAction"}
   {"@id" "schema:LinkRole"}
   {"@id" "schema:BroadcastService"}
   {"@id" "schema:PronounceableText"}],
  "schema:rangeIncludes" [{"@id" "schema:Text"} {"@id" "schema:Language"}],
  "schema:source" {"@id" "https://github.com/schemaorg/schemaorg/issues/2382"}}]
```

### PR DESCRIPTION
**Pull request by zonotope**
This patch correctly resolves the subject id for iri maps in the object position. Without this patch, _all_ subjects with the property will match a given pattern with an iri map in the object position, instead of only the subjects that have the specified object for that property.

We did have a test for iri maps, but that test was passing when it shouldn't have been because we had only one subject in our test data with the given property ("schema:isBasedOn"), so I added another one and updated the tests accordingly.

I also took the time to add some (but definitely not all) of the indexing optimizations I found while exploring presistent-sorted-set, and I also cleaned up our test/dev logging configuration so the repl isn't flooded with trace logs. 

Fixes #528

### CODE REVIEW

File: src/fluree/db/indexer/default.cljc [hunk 1]
```diff
@@ -292,10 +291,10 @@
...
                (if (and child
                         (index/descendant? node child))     ; all of a resolved
-                 ; branch's children
-                 ; should be at the top
-                 ; of the stack
-                 (recur (conj child-nodes child)
+                                                            ; branch's children
+                                                            ; should be at the top
+                                                            ; of the stack
+                 (recur (conj child-nodes (index/unresolve child))
```
Comment (bplatz) [inline src/fluree/db/indexer/default.cljc:297]: I assume this is to allow for GC... if so, could this be part of the very large ledger reindexing slow-downs we saw in v2?

Comment (zonotope) [reply src/fluree/db/indexer/default.cljc:297]: Yes, and yes. Without this, the whole index tree sits on the database during a reindex.

**Review by dpetran — APPROVED** [review]
:+1: These look like good changes to me.

If I understand it correctly, fixing the problem with id-map lookups is solved in `where/resolve-flake-range`. Before we were trying to get the ::iri value of the ::val, but ::iri is actually a key on the `o-cmp`.

**Comment by zonotope** [comment]
> +1 These look like good changes to me.
> 
> If I understand it correctly, fixing the problem with id-map lookups is solved in `where/resolve-flake-range`. Before we were trying to get the ::iri value of the ::val, but ::iri is actually a key on the `o-cmp`.

Exactly

**Pull request merged**